### PR TITLE
Feature/compute score

### PR DESCRIPTION
**Pull request by alvarogonzalezDEV**

### CODE REVIEW

File: src/components/compute-score.html [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+          /** candidate information */
+          userData: Object,
+          /** input object where store the test information  */
+          statusData: Object,
+          /** input object that stores the answers */
+          correctQuestions: Object,
+          /** output object to generate the mail template */
+          scoreData: {
+            type: Object,
+            value: () => { }
```
Comment (edu-aguilar) [inline src/components/compute-score.html]: careful!!! es6 understand this as a function:  () => { }  if you want to return a Object with an arrow function, you must type: () => ({})

File: src/components/compute-score.html [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                  question.userAnswer = selectedOptionAswered;
+                break;  
+              default:
+                break;
+            }   
+            scoreData.questions.push(question);
+        }
+        scoreData.correctPercentage = Math.round((scoreData.correctCount * 100 ) / scoreData.totalCount);
+        this.set('scoreData', scoreData);
+      }
```
Comment (edu-aguilar) [inline src/components/compute-score.html]: good code, but this function is toooo long. For example, try to initialize the storeData object within a function.

File: src/components/compute-score.html [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      _builtTemplate(){
+        
+       let scoreData = this._inizializeScoreData();
+
+        for (var i = 0; i < this.statusData.quizData.questions.length; i++) {
+          let question = {}
+          question.id = this.statusData.quizData.questions[i].id;
+          question.statement = this.statusData.quizData.questions[i].question;
+          question.type = this.statusData.quizData.questions[i].type;
+          question.correct = false;
```
Comment (pepocivs) [inline src/components/compute-score.html]: I think is better declare a populated object instead statement by statement, somethink like `let question = {
id: this.statusData.quizData.questions[i].id,
statement: this.statusData.quizData.questions[i].question
}`

or, use the same keyNames and `let question = this.statusData.quizData.questions[i]` and then populate custom properties as a `question.correct = false`

You'll save some lines

File: src/components/compute-score.html [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        for (var i = 0; i < this.statusData.quizData.questions.length; i++) {
+          let question = {}
+          question.id = this.statusData.quizData.questions[i].id;
+          question.statement = this.statusData.quizData.questions[i].question;
+          question.type = this.statusData.quizData.questions[i].type;
+          question.correct = false;  
+          const correctOptionAswered = this.correctQuestions[question.id];
+          const selectedOptionAswered = this.statusData.answeredQuestions[question.id];
+
+            switch (this.statusData.quizData.questions[i].type) {
```
Comment (pepocivs) [inline src/components/compute-score.html]: here question.type is shorter

File: test/components/compute-score.html [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+</head>
+
+<body>
+
+  <test-fixture id="computeScoreTest">
+    <template>
+      <compute-score
+       user-data='{ "name": "candidato1", "surname": "ape2", "email": "candidato@soprasteria.com"}'
+       correct-questions='{"1": "1", "2": "1" , "3": "2", "5" : ["2","3"], "6" : ["2","3"]}'
+       status-data='{ "answeredQuestions": {"1": "3", "2": "1" , "3": "2", "4": "estilos en cascada", "5" : ["2","3"], "6" : ["1","3"] },
```
Comment (pepocivs) [inline test/components/compute-score.html]: BTM is good, but i think we have to create "mock" folder inside test folder with all of mocked jsons

**Pull request merged**